src/addrman.cpp

```diff
@@ -699,3 +700,100 @@ std::vector<bool> CAddrMan::DecodeAsmap(fs::path path)
     }
     return bits;
 }
+
+void CAddrMan::ResetI2PPorts()
+{
+    for (int bucket = 0; bucket < ADDRMAN_NEW_BUCKET_COUNT; ++bucket) {
+        for (int i = 0; i < ADDRMAN_BUCKET_SIZE; ++i) {
+            const auto id = vvNew[bucket][i];
+            if (id == -1) {
+                continue;
+            }
+            auto it = mapInfo.find(id);
+            if (it == mapInfo.end()) {
+                return;
```

Comment:
(Asserts run fine for me in my testing so far).

Reply:
Asserts look [scary](https://github.com/bitcoin/bitcoin/pull/22112#discussion_r662245348) to me. If those have to be asserts then I would deem this too risky for 22.0. Maybe I am pessimistic but I imagine 22.0 nodes not being able to start because I overlooked something in the I2P ports fiddling which is definitely not critical for node operation :bomb: :skull_and_crossbones: